### PR TITLE
 Transpile Network Code

### PR DESCRIPTION
**Pull request by isiko**
# Description
The `info.metadude.android.eventfahrplan.network` package still had some `.java` files. As the standard language for Android apps is now Kotlin, I transpiled them to Kotlin. I used the transpiling tool, provided by Android Studio and made some cleanups afterwards.

# Tested on (with all build flavors):
- Phone (Pixel 2), API lvl 24 Emulator
- Phone (Pixel 2), API lvl 33 Emulator

Relates to #448

### CODE REVIEW

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        exceptionMessage = getExceptionMessage(exception.cause, exception.message) ?: ""
+    }
+
+    private fun customizeExceptionMessage(exception: SSLException) {
+        if (exception is SSLHandshakeException && Build.VERSION.SDK_INT == Build.VERSION_CODES.N) {
+            // See https://github.com/EventFahrplan/EventFahrplan/issues/431
+            exceptionMessage += """
+                
+                
+                Please note that server certificates using elliptic curves with a length > 256 bits are not supported on Android 7.0. This might cause this error.
```
Comment (johnjohndoe) [inline network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt:185]: Please improve readability (line breaks) while keeping the original output formatting in the error dialog.

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    private lateinit var task: FetchFahrplanTask
+    private lateinit var listener: (fetchScheduleResult: FetchScheduleResult) -> Unit
+    fun fetch(okHttpClient: OkHttpClient, url: String?, eTag: String?) {
+        task = FetchFahrplanTask(okHttpClient, logging, listener)
+        task.execute(url, eTag)
+    }
+
+    fun cancel() {
+        if (this::task.isInitialized) {
```
Comment (johnjohndoe) [inline network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt:32]: `this` can be removed (check all locations).

**Review by johnjohndoe — CHANGES_REQUESTED** [review]

**Comment by isiko** [comment]
Should be all done now

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            if (statusCode == 401) {
+                return HttpStatus.HTTP_WRONG_HTTP_CREDENTIALS
+            }
+            return if (statusCode == 404) {
+                HttpStatus.HTTP_NOT_FOUND
+            } else HttpStatus.HTTP_COULD_NOT_CONNECT
+        }
+
+        eTagStr = response.header("ETag") ?: ""
+        if (eTagStr.isNotEmpty()) {
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt]: You can reverse the if condition and get rid of the ugly `!`.

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        fun getExceptionMessage(cause: Throwable?, message:String?): String? =
+            if (cause == null) message else getExceptionMessage(cause.cause, cause.message)
+
+        exceptionMessage = getExceptionMessage(exception.cause, exception.message).orEmpty()
+    }
+
+    private fun customizeExceptionMessage(exception: SSLException) {
+        if (exception is SSLHandshakeException && Build.VERSION.SDK_INT == Build.VERSION_CODES.N) {
+            // See https://github.com/EventFahrplan/EventFahrplan/issues/431
+            ("${exceptionMessage}\n\n\n\t" +
```
Comment (johnjohndoe) [inline network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt:180]: Are the curly braces needed?
Why three `\n`?
Why the `\t`?
Please compare on screen with the old version. You can trigger the dialog manually by changing a bit of code.

Comment (isiko) [reply network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt]: Sorry for that, I was just confused by the stuff Android Studio transpiled. Should be good now

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+    fun setListener(listener: (fetchScheduleResult: FetchScheduleResult) -> Unit) {
+        onFetchScheduleResult = listener
+        if (::task.isInitialized) {
+            task.setListener(listener)
+        }
+    }
+}
+
+class FetchFahrplanTask(
```
Comment (johnjohndoe) [inline network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt:43]: Did you remove the `internal` intentionally?

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+            if (eTag.isNotEmpty()) {
+                addHeader("If-None-Match", eTag)
+            }
+        }
+
+        val e = SSLException("Hallo das ist ein Test")
+        setExceptionMessage(e)
+        customizeExceptionMessage(e)
+        e.printStackTrace()
+
```
Comment (johnjohndoe) [inline network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt:118]: Left-over from debugging?

**Review by johnjohndoe — CHANGES_REQUESTED** [review]

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt (hunk 5 above)
Comment (isiko) [reply network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt]: Nope, I'm just a bit Stupid :sweat_smile:

File: network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt (hunk 6 above)
Comment (isiko) [reply network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt]: Should be removed now :facepalm:

**Review by johnjohndoe — APPROVED** [review]
Thank you for the conversion! LGTM :+1:

**Pull request merged**